src/demo/components/redux-example.ts

```diff
@@ -37,39 +39,46 @@ template.innerHTML = `
 `
 
 class ReduxExample extends connect(store)(HTMLElement) {
+  private _counter: CounterElement
+  private _clicksSpan: HTMLElement
+  private _valueSpan: HTMLElement
+  private _didLoadSpan: HTMLElement
   constructor() {
     super();
 
     // Stamp the template.
-    let shadowRoot = this.attachShadow({mode: 'open'});
-    this.shadowRoot.appendChild(document.importNode(template.content, true));
+    let shadowRoot = this.attachShadow({mode: 'open'})!;
+    shadowRoot.appendChild(document.importNode(template.content, true));
 
     // Cache some elements so that you don't qsa all the time.
-    this._counter = shadowRoot.querySelector('counter-element');
-    this._clicksSpan = shadowRoot.getElementById('clicksSpan');
-    this._valueSpan = shadowRoot.getElementById('valueSpan');
-    this._didLoadSpan = shadowRoot.getElementById('didLoadSpan');
+    this._counter = shadowRoot.querySelector('counter-element')! as CounterElement;
```

Comment:
Instead of this cast, can extend the HTMLElementTagNameMap in counter-element

Reply:
I removed the extension in https://github.com/Polymer/pwa-helpers/pull/39/commits/96251f0794a30159a39f6c42f5354d700af27c80 because this extension (which is intended for the demo in this repo only) was adding `'counter-element'` to the generated typings in connect-mixin.d.ts:

```js
export declare const connect: /*...*/(baseElement: T) => {
  new (...args: any[]): {
    /*...*/
    querySelector<K extends "object" | "font" | /*...*/ | "xmp" | "counter-element">(selectors: K): HTMLElementTagNameMap[K] | null;
    /*...*/
  };
} & T;
export {};
```

which is consumed by apps using this library which may not have a `counter-element` and causes a TS compile error for the app:

```
../pwa-helpers/connect-mixin.d.ts:156:1260 - error TS2536: Type 'K' cannot be used to index type 'HTMLElementTagNameMap'.
```

Ideally there's a way to remove this typing, but I think this is just how TS creates typings for mixin functions.

Reply:
Oh I see, I missed that this was in the demo subdirectory.

Reply:
Also, ouch on that typings expansion. That looks like it's related to https://github.com/Microsoft/TypeScript/issues/27171